_posts/en/newsletters/2019-05-14-newsletter.md

```diff
@@ -0,0 +1,485 @@
+potentially improving privacy.  After ordering, the two hashes will be
+hashed together with the prefix tag "TapBranch".  As this merkle tree
+only has two leaves, the resultant hash is the merkle root.
+
+    FIXME: diagram of tree
+
+This merkle tree only covers two of the HTLC's possible end states.  For
+the third case where Alice and Bob mutually agree on a spend, they can
+use signature aggregation using something like MuSig.  As in the
+multisig section above, they work together to create a single pubkey,
+called the *Taproot internal key.*
+
+The merkle root and the internal key are then hashed together (prefix
+tag, "TapTweak") and the resultant digest is used as a private key from
+which a pubkey is derived, this value being known as the *tweak.*  The
+tweak pubkey is added to the internal key (the same operation used in
+[BIP32][] HD key derivation) in order to derive the *taproot output
+key---*the key that is used in any bech32 addresses and scriptPubKeys
+that pay these three conditions.
+
+    FIXME: diagram of tweaking, similar to one from BitMex article
+
+When it comes time to spend this money, either Alice or Bob can provide
+the script they want to use, the data needed by it (a signature and,
+sometimes, a hash preimage), the Taproot internal key, and the hash of
+the script they didn't use.  Alternatively, Alice and Bob work together
+to use signature aggregation (after accounting for the tweak) to provide
+a signature in combination with the same single-key, single-signature
+spending pattern described in the previous subsections.  As long as the
+data they provide in either case is correct, the spend will be accepted.
+
+    FIXME: table example of HTLC cost in current P2WSH and taproot for
+    each of the three possible outcomes
+
+    P2WSH
+      Output: 8+1+34
+      1.preimage, digest, A pubkey, B pubkey, OP_IF/ENDIF, Alice sig
+      2. digest, A pubkey, B pubkey, IF/ENDIF, Bob sig
+      3. N/A (use one of the above)
+
+    Taproot
+      Output: 8+1+35
+      1. preimage, digest, A pubkey, A sig; internal key; leaf hash
+      2. B pubkey, B sig; internal key; leaf hash
+      3. AB sig
+
+Because we chose to use a small tree in the above example, we will
+describe some things that would've been clearer with a larger example.
+First, if you have any unpaired nodes in your tree, your software will
+be responsible for pairing them with something, e.g a 32-byte random
+number.  Additionally, the maximum depth of a tree is 32 rows, allowing
+a tree to contain a maximum of around four billion scripts.  However,
+only one of those scripts would appear in any spending transaction and
+only 32 hashes directly related to the merkle tree would need to be
+generated to prove the script was part of the tree---this means that more
+complex scripts than our simple HTLC can gain much larger efficiency
+improvements than seen above (see an article focused on [MAST][] for
+more information).
+
```

Comment:
This is great info, but if we go with my previous comment about not getting into MuSig details, then I don't think this fits very well in this section.

I was thinking that, when we finish the 24 weeks of bech32, we might keep the long-term focus section and convert it to describing a detail about schnorr/taproot/tapscript each week.  I think this would work well in something like that.  Alternatively, if you don't want to wait three months, you could post a blog (I'd be happy to help, if you want).

Reply:
Sounds good and either approach works for me.